recipes/cmake/3.x.x/conanfile.py

```diff
@@ -54,6 +79,12 @@ def package(self):
         cmake.install()
         tools.rmdir(os.path.join(self.package_folder, "doc"))
 
+    def package_id(self):
+        self.info.options.with_openssl = self._with_openssl
+
+        del self.info.settings.compiler
+        del self.info.settings.build_type
```

Comment:
@jgsogo @SSE4 
By not removing compiler and build_type, at least 50 packages of cmake will be built for each version.
```suggestion
```

Reply:
Yes. I understand that the desired scenario would be to remove these settings but to be able to tell C3i that we want to build the `Release` build.... that way, when someone retrieves the package from CCI it will be the _good_ one.

This is not possible right now, the only alternative is to build them all 😞

Reply:
I'll apply both suggestions then.

Is there already an issue open on github about this?

Reply:
No, none of the two issues here, but both are really hard to tackle.

From the CCI perspective we don't want to run a different pipeline based on custom attributes and Conan right now doesn't have a feature to classify packages (and IMO we don't need it yet). The other issue: being able to assign different _build_ profiles to different branches of the graph, requires a new graph model, something we are working for Conan 2.0, but it will require time to implement.

So, for the time being, we are building ~100 packages for each CMake version even though we will probably use only a dozen of them

Reply:
Can't you just add an entry in the top-level yml file for telling cci which permutations to build? If 'compiler' and 'build_type' are removed from the exported 'package_id', no consumer should ever need to rebuild this package, no matter what profile they use. Or am I missing something?

Reply:
We can think about different alternatives to tell CCI which configurations to build and probably it is about something in the `config.yml` file as you have suggested. OTH, if we remove `build_type` from the package ID we cannot guarantee (at this moment) that CCI will choose `Release` build (this was the issue originating this PR, I think).

Even though these things might look almost trivial, they are not as straightforward as they look like, and there are other things in the backlog with higher priority. We would really want to address all these issues, but we need to go step by step.